### DISCUSSION TITLE
Update parent package versions on same 'deploy'

### DISCUSSION THREAD

**Discussion by rcompton78** [jscutlery/semver#484] [Ideas]
Just a suggestion:

Add a flag to have the version act similar to lerna when there is package-a that depends on package-b and you have changes to package-b.  This flag would update package-a to use the new package-b version.  I realise if package-a has ~/^ as the version it most likely will get the new package on install but as a flag it would be nice.  

Somewhat going a long with this is it possible to have one git commit for X package changes instead of X commits?

Thanks for all the hard work.

**Comment by edbzn** [comment]
Does the [`--trackDeps`](https://github.com/jscutlery/semver#tracking-dependencies) option is what your are looking for?

**Reply by rcompton78** [reply]
That works perfect.   I tried it out originally but I must have missed something in the setup.  Likely a project.json missing trackDeps

**Reply by rcompton78** [reply]
Actually after testing a bit more I think it is either a bug or missing an option.  When you use trackDeps it ups the version of all the dependent packages correctly but does not update the actual package version in the dependencies[] in package.json.  Ex;

PackageA (1.1.0)  
dependencies: [ PackageB@1.0.0]  

PackageB (1.0.0)

Make a oatch change to packageB and semver does
PackageA(1.011)
dependencies: [PackageB@1.0.0]

PackageB(1.0.1)

As I mentioned mostly likely if you have ~/^ it will pick up the patch/minor changes as you want but Lerna allows you to auto update the versions in the deps[] to keep things synced or work with exact versions.

**Reply by edbzn** [reply]
@rcompton78 Because semver assumes that you are using path-alias for importing dependencies, not an external npm package. It follows the Nx philosophy which enforces the single version policy.
